### ISSUE TITLE
Scrolling and visibility

### ISSUE THREAD

**Issue by atkinsonline** [SKLn-Rad/Xam.Plugin.Webview#23]
Hi Ryan. I have a problem which at first I thought was a general Xamarin problem but I tried replacing your webview with the standard webview and the problem went away.  What I'm trying to achieve is the following.
A header at the top of the page containing a button.  The webview taking up the body of the page and a footer at the bottom of the page also containing a button.  The idea is that to start with the footer is invisible and when I tap the button in the header, the footer appears and when I tap the button in the footer the footer disappears. So I can toggle on and off seeing the footer.
If I set the footer to be invisible at startup time, it never appears when I set it to visible.  Also, scrolling of the webview stops when I make the footer visible. Until then scrolling works fine.
If I set the footer to be visible at startup, scrolling is fine until I tap the footer button and make the footer invisible.  Then webview scrolling stops again
I read a bit about problems with scrolling and making things invisible so I tried leaving the footer always visible but toggling its height between 0 and 100.  As soon as the footer height is zero, the webview scrolling stops and the footer never reappears.
I expected this to be a weird xamarin problem but if I use a normal Xamarin webview it all works as I'd expect.
I've included some of my code below to show the setup. I'm new to Xamarin so it may be a simple problem of how I'm creating or initialising the views but I've tried everything I can think of.
Thanks, John.

`        public WebViewPage()
        {

            WebViewContentType contentType = WebViewContentType.Internet;
            //WebViewContentType contentType = WebViewContentType.StringData;

            //string url = getURL(contentType);
            string url = "http://www.bbc.co.uk";

            WebView = new FormsWebView()
            {
                VerticalOptions = LayoutOptions.FillAndExpand,
                HorizontalOptions = LayoutOptions.FillAndExpand,
                ContentType = contentType,
                Source = url
            };
            WebView.RegisterCallback("OnLoadPhotoHandlerCS", OnLoadPhotoHandlerCS);

            //var webview = new Xamarin.Forms.WebView
            //{
            //    HorizontalOptions = LayoutOptions.Center,
            //    VerticalOptions = LayoutOptions.CenterAndExpand,
            //    Source = url
            //};

            var btnShowFooter = new Button
            {
                Text = "Show Footer",
                Font = Font.SystemFontOfSize(NamedSize.Large),
                BorderWidth = 1,
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.CenterAndExpand
            };
            btnShowFooter.Clicked += OnShowFooterClicked;

            var btnClose = new Button
            {
                Text = "Close",
                Font = Font.SystemFontOfSize(NamedSize.Large),
                BorderWidth = 1,
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.CenterAndExpand
            };
            btnClose.Clicked += OnCloseClicked;

            var header = new StackLayout
            {
                BackgroundColor = Color.Green,
                HeightRequest = 50,
                HorizontalOptions = LayoutOptions.FillAndExpand,
                VerticalOptions = LayoutOptions.Start,
                Children = {
                        btnShowFooter
                    }
            };

            footer = new StackLayout
            {
                BackgroundColor = Color.LightCyan,
                HeightRequest = 100,
                HorizontalOptions = LayoutOptions.FillAndExpand,
                VerticalOptions = LayoutOptions.End,
                //IsVisible = true,
                Children = {
                        btnClose
                    }
            };

            Content = new StackLayout
            {
                VerticalOptions = LayoutOptions.FillAndExpand,
                HorizontalOptions = LayoutOptions.FillAndExpand,
                BackgroundColor = Color.Red,
                Children = {
                        header,
                        WebView,
                        //webView,
                        footer
                    }
            };

            WebView.OnNavigationStarted += OnNavigationStarted;
            WebView.OnNavigationCompleted += OnNavigationComplete;
            WebView.OnContentLoaded += OnContentLoaded;
            WebView.OnJavascriptResponse += OnJavascriptResponse;

            NavigationPage.SetHasNavigationBar(this, false);
            this.Appearing += Content_Appearing;

        }

        private void OnShowFooterClicked(object sender, EventArgs e)
        {
            Debug.WriteLine("OnShowFooterClicked");

            this.footer.IsVisible = true;
            this.footer.HeightRequest = 100;

            //var mediaChooserPage = new MediaChooserPage();
            //await Navigation.PushModalAsync(mediaChooserPage);

        }

        private void OnCloseClicked(object sender, EventArgs e)
        {
            Debug.WriteLine("OnCloseClicked");

            //this.footer.IsVisible = false;
            this.footer.HeightRequest = 0;

        }
`

**Comment by SKLn-Rad** [comment]
Hey John! Sorry about the slow reply, I'm just preparing to go on holiday
so I've been busy handing over work :(

To the point, I don't have access to any real devices at the moment so I
have had to test with UWP and it seems to work correctly there! What
platform did you test this on? Android may differ from native due to native
defaulting to software whereas mine is set up to use hardware acceleration,
so my first thoughts are that and to change that behaviour using the
OnControlChanging event.

Its great to see so many get into Xamarin; I included a sample which does
what you want but uses translation instead of heightrequest - Its faster as
it doesn't perform a full layout cycle + animations! :D
Take it you don't work for the BBC then as you used that URL? if so we may
have met in the past when I was there as a tester :p

If you need any further help, honestly man just pop me a message on
whatever medium you want, just let me know.

Slack: https://xamarinchat.herokuapp.com
Discord: https://discord.gg/8Q8J77J
LinkedIn: linkedin.radsrc.com

- Ryan

**Comment by SKLn-Rad** [comment]
Note: I forgot to set the margin on the webview when the footer is
displayed on that example, you would just do that on
the OnFooterDisplayChange callback!
So ignore the overlaying footer :)

**Comment by atkinsonline** [comment]
Hi Ryan, thank you for the reply and no problem at all with the delay.  It's great that you respond very quickly and put in a lot of effort trying to help.
I should have said, it's Android.  It was just a test really of trying to do something to see how it would look.  I decided to abandon that and use a pop up modal window that slides in from the bottom rather than having a hidden view that appears.  After reading a lot about modal form popups (that do not fill the screen) I settled on rg.Plugins.popup that works really well.
I worked a lot in London one time but never for the BBC, I now live in Devon. I just used the BBC website as an example.
Have a great holiday.
John.
